contracts/prize-strategy/multiple-winners/MultipleWinners.sol

```diff
@@ -64,33 +102,42 @@ contract MultipleWinners is PeriodicPrizeStrategy {
   function _distribute(uint256 randomNumber) internal override {
     uint256 prize = prizePool.captureAwardBalance();
 
-    // main winner is simply the first that is drawn
-    address mainWinner = ticket.draw(randomNumber);
-
-    // If drawing yields no winner, then there is no one to pick
-    if (mainWinner == address(0)) {
+    if (IERC20Upgradeable(address(ticket)).totalSupply() == 0) {
       emit NoWinners();
       return;
     }
 
-    // main winner gets all external ERC721 tokens
-    _awardExternalErc721s(mainWinner);
-
-    address[] memory winners = new address[](__numberOfWinners);
-    winners[0] = mainWinner;
+    bool _carryOverBlocklistPrizes = carryOverBlocklist;
 
+    // main winner is simply the first that is drawn
+    uint256 numberOfWinners = __numberOfWinners;
+    address[] memory winners = new address[](numberOfWinners);
     uint256 nextRandom = randomNumber;
-    for (uint256 winnerCount = 1; winnerCount < __numberOfWinners; winnerCount++) {
+    uint256 winnerCount = 0;
+    uint256 retries = 0;
+    uint256 _retryCount = blocklistRetryCount;
+    while (winnerCount < numberOfWinners) {
+      address winner = ticket.draw(nextRandom);
+
+      if (!isBlocklisted[winner]) {
+        winners[winnerCount++] = winner;
+      } else if (++retries >= _retryCount) {
+        emit RetryMaxLimitReached(numberOfWinners);
```

Comment:
Should check if no winners have been selected when try retry limit has been reached.

If no winners have been selected and the retry limit is reached the `NoWinners();` event should be emitted.

```
if (!isBlocklisted[winner]) {
  winners[winnerCount++] = winner;
} else if (++retries >= _retryCount) {
  emit RetryMaxLimitReached(numberOfWinners);
  if(winnerCount == 0) {
    emit NoWinners();
  }
  break;
}
```

Reply:
Great call!

Also, I realize we should emit the `winnerCount` here, not the `numberOfWinners`.  We can infer the number of winners just as we do with the retry count, but the *actual* winner count is unique and should be recorded.